### PR TITLE
[gRPC/Tests] Add some unittests for gRPC source/sink

### PR DESCRIPTION
**Pull request by dongju-chae**
This patch adds some unittests for gRPC source/sink, including
- basic operations & property
- multi-pieline examples

Signed-off-by: Dongju Chae <dongju.chae@samsung.com>

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -105,6 +106,13 @@ Depends: nnstreamer, ${shlibs:Depends}, ${misc:Depends}
 Description: NNStreamer Flatbuf converter/decoder support
  This package allows to pack/unpack tensor streams to/from flatbuf.
 
+Package: nnstreamer-grpc
+Architecture: amd64
+MUlti-Arch: same
```
Comment (leemgs) [inline debian/control:111]: **s/MU/Mu/**

Comment (dongju-chae) [reply debian/control:111]: Ohh.. I will fix this in another PR.

File: tests/nnstreamer_grpc/runTest.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+do
+  callCompareTest original_${i}.log result_${i}.log 1-${i} "Compare 1-${i}" 1 0
+done
+
+rm result_*.log
+
+# tensor_sink (server) --> tensor_src (client)
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=${NUM_BUFFERS} ! video/x-raw,width=640,height=480,framerate=5/1 ! tensor_converter ! tensor_sink_grpc server=true" 2 0 0 $PERFORMANCE &
+sleep 1
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} tensor_src_grpc num-buffers=${NUM_BUFFERS} server=false ! 'other/tensor,dimension=(string)3:640:480,type=(string)uint8,framerate=(fraction)5/1' ! multifilesink location=result_%1d.log" 2 0 0 $PERFORMANCE
```
Comment (gichan-jang) [inline tests/nnstreamer_grpc/runTest.sh:43]: How about adding a test for other/tensors?

File: tests/nnstreamer_grpc/unittest_grpc.cc [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+  src = gst_bin_get_by_name (GST_BIN (test_data.pipeline), "src");
+  ASSERT_TRUE (src != NULL);
+
+  g_object_get (src, "silent", &silent, NULL);
+  EXPECT_TRUE (silent);
+
+  g_object_get (src, "server", &server, NULL);
+  EXPECT_EQ (server, DEFAULT_SERVER);
+
+  g_object_get (src, "host", &host, NULL);
```
Comment (gichan-jang) [inline tests/nnstreamer_grpc/unittest_grpc.cc:182]: need to free host.
The rest of the tests need to be modified also.

File: tests/nnstreamer_grpc/runTest.sh (hunk 2 above)
Comment (dongju-chae) [reply tests/nnstreamer_grpc/runTest.sh:43]: Got it. Thanks.

File: tests/nnstreamer_grpc/unittest_grpc.cc (hunk 3 above)
Comment (dongju-chae) [reply tests/nnstreamer_grpc/unittest_grpc.cc:182]: Got it. Thanks.

**Review by gichan-jang — APPROVED** [review]
LGTM

File: tests/nnstreamer_grpc/runTest.sh [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+# @brief SSAT Test Cases for NNStreamer
+#
+
+if [[ "$SSATAPILOADED" != "1" ]]; then
+    SILENT=0
+    INDEPENDENT=1
+    search="ssat-api.sh"
+    source $search
+    printf "${Blue}Independent Mode${NC}
+"
```
Comment (wooksong) [inline tests/nnstreamer_grpc/runTest.sh:16]:
```suggestion
    printf "${Blue}Independent Mode${NC}"
```

File: tests/nnstreamer_grpc/unittest_grpc.cc [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+ */
+TEST (nnstreamer_grpc, check_existence)
+{
+  GstPlugin *plugin;
+  GstElementFactory *factory;
+
+  plugin = gst_registry_find_plugin (gst_registry_get (),
+      NNS_GRPC_PLUGIN_NAME);
+  EXPECT_TRUE (plugin != NULL);
+  gst_object_unref (GST_OBJECT (plugin));
```
Comment (wooksong) [inline tests/nnstreamer_grpc/unittest_grpc.cc:31]:
```suggestion
  gst_object_unref (plugin);
```
Casting to ```GST_OBJECT``` looks not necessary.
https://github.com/GStreamer/gstreamer/blob/master/gst/gstplugin.c

Comment (dongju-chae) [reply tests/nnstreamer_grpc/unittest_grpc.cc:31]: Right, thanks.

File: tests/nnstreamer_grpc/runTest.sh (hunk 4 above)
Comment (dongju-chae) [reply tests/nnstreamer_grpc/runTest.sh:16]: Oh, thanks.

**Review by myungjoo — APPROVED** [review]

**Pull request merged**